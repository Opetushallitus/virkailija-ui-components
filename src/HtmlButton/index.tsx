import * as React from 'react';

const HtmlButton = React.forwardRef<
  HTMLButtonElement,
  React.ComponentProps<'button'>
>(({ type = 'button', ...props }, ref) => {
  return <button type={type} ref={ref} {...props} />;
});

export type HtmlButtonProps = React.ComponentProps<'button'>;

export default HtmlButton;
