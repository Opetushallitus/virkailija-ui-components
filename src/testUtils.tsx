import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createTheme } from './createTheme';

const theme = createTheme();

const Providers = ({ children }: { children: React.ReactNode }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const dummyDiv = document.createElement('div');
dummyDiv.setAttribute('data-nothing', 'true');

// Return rootElement from render() for easier snapshot testing
const customRender = (ui: React.ReactElement, options: any = {}) => {
  const result = render(ui, { wrapper: Providers, ...options });
  return {
    ...result,
    rootElement: result.container.firstChild || dummyDiv,
  };
};

export * from '@testing-library/react';

export { customRender as render };
