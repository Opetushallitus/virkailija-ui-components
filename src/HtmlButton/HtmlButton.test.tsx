import React from 'react';
import { render } from '../testUtils';
import HtmlButton from './index';

it('Renders a HtmlButton', () => {
  const { rootElement } = render(<HtmlButton>test</HtmlButton>);
  expect(rootElement).toMatchSnapshot();
});
