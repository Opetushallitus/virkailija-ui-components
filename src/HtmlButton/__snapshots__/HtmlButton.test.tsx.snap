// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a HtmlButton 1`] = `
<button
  type="button"
>
  test
</button>
`;
